hash_practice/exercises.py

```diff
@@ -1,19 +1,48 @@
+from collections import defaultdict
 
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n*m) because m depends on how long the string is, so that is variable 
+        Space Complexity: O(n) because we're creating an array that can be, worst case, as long as the initial array
     """
-    pass
+    anagram_dict = defaultdict(list)
+    result = []
+
+    for word in strings:
+        for char in word:
+            result.append(char) #gets all the characters in a word, ie "e", "a", "t"
+        result.sort()
+        letter_tuple = tuple(result)
+        anagram_dict[letter_tuple].append(word)
+        result = []
+
+    for item in anagram_dict.values():
+        result.append(item)
+    
+    return result
+    
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)

```

Comment:
This would be fine if we did not then call `elements.sort` on line 40, which is going to take O(n*log(n)) time, and still needs to be accounted for in your calculation like in the anagrams problem, and thus will be the final complexity.